src/validation/statement.rs

```diff
@@ -521,10 +521,11 @@ fn validate_reference<T: AnnotationMap>(
                         !qualified_name.starts_with(pou)
                             && !qualified_name.starts_with(container)
                             && !context.index.is_init_function(pou)
+                            //Hack: Avoid internal check here because of the super call
+                            && !location.is_internal()
```

Comment:
```suggestion
                        !(
                            qualified_name.starts_with(pou)
                                || qualified_name.starts_with(container)
                                || context.index.is_init_function(pou)
                                //Hack: Avoid internal check here because of the super call
                                || location.is_internal()
                         )
```